pkg/engine/task/task.go

```diff
@@ -126,6 +130,22 @@ func newPipe(task *v1beta1.Task) (Tasker, error) {
 	}, nil
 }
 
+func newToggle(task *v1beta1.Task) (Tasker, error) {
+	// validate if resources are present
+	if len(task.Spec.ResourceTaskSpec.Resources) == 0 {
+		return nil, errors.New("task validation error: toggle task has an empty resource list. if that's what you need, use a Dummy task instead")
+	}
+	// validate if the parameter is present
+	if len(task.Spec.ToggleTaskSpec.Parameter) == 0 {
+		return nil, errors.New("task validation error: Missing parameter to evaluate the Toggle Task")
+	}
+	return ToggleTask{
+		Name:      task.Name,
+		Resources: task.Spec.ResourceTaskSpec.Resources,
```

Comment:
same here:
```suggestion
		Resources: task.Spec.ToggleTaskSpec.Resources,
```

Reply:
I've basically undone this change, as the `ToggleTaskSpec` doesn't actually have the `Resources` field (as this would lead to a duplicate field in the TaskSpec... )

Reply:
Hm, I see what you did there. However, now `ToggleTaskSpec` itself is not self-contained like the other task specs: it only has the `Parameter` but not the `Resources` 🤔

Reply:
Yeah. I don't like it either, but it's the only way for now, as we **can not** have fields with the same name in different TaskSpec types.

From the `operatorversion_types.go`:
```
// TaskSpec embeds all possible task specs. This allows us to avoid writing custom un/marshallers that would only parse
// certain fields depending on the task Kind. The downside of this approach is, that embedded types can not have fields
// with the same json names as it would become ambiguous for the default parser. We might revisit this approach in the
// future should this become an issue.
```

So.... Maybe this is the point where we need to approach this?

Reply:
Fun fact: I wrote that comment 🤣 And yes, we might approach the time when a custom parser is needed. Or not 🤷‍♂